bigquery_etl/cli/query.py

```diff
@@ -847,14 +861,23 @@ def _run_query(
     dataset_id,
     query_arguments,
     addl_templates: Optional[dict] = None,
+    billing_project: Optional[str] = None,
 ):
-    """Run a query."""
+    """Run a query.
+
+    project_id is the default project to use with table/view/udf references in the query that
+    do not have a project id qualifier.
+    billing_project is the project to run the query in for the purposes of billing and
+    slot reservation selection.  This is project_id if billing_project is not set
+    """
     if dataset_id is not None:
         # dataset ID was parsed by argparse but needs to be passed as parameter
         # when running the query
-        query_arguments.append("--dataset_id={}".format(dataset_id))
+        query_arguments.append(f"--dataset_id={dataset_id}")
```

Comment:
I think `--dataset_id` probably shouldn't be set if `billing_project` is specified.

Reply:
Yes, `SET @@dataset_id` takes precedence so this shouldn't be set if it doesn't do anything

Reply:
Oh wait, I forgot I also suggested we only do the session logic if `billing_project` was different than `project_id`, so if they're the same project then the `--dataset_id` arg should still be added.

Reply:
Oh right, I'll fix this in another pr when I add `--billing_project` to something